### PR TITLE
Naf obj suppress null #178

### LINKED ISSUE #178: Add support for null results suppression to prevent unnecessary data being sent
Request is for an additional source configuration that when enabled will not transmit null results. This will prevent unnecessary events from streaming to the platform and may allow for higher performance on the machines running the extension sources. 

I am specifically asking for objectRecognition here but I am guessing this could be useful to other ext-src projects as well.

### PR DESCRIPTION
**Pull request by namirfa**
Closes #178 

This enhancement adds a new general config option, `suppressNullValues`. If this flag is set to true, then any polling data with no recognitions will not be sent to the source.

Also pulled some duplicated test code out and put it into NeuralNetTestBase.

All tests pass as expected, and build is successful.

### CODE REVIEW

File: objectRecognitionSource/README.md [hunk 1]
```diff
@@ -149,7 +149,8 @@ The Configuration document may look similar to the following example:
           "general": {
              "pollTime": 3000,
              "maxRunningThreads": 5,
-             "maxQueuedTasks": 10
+             "maxQueuedTasks": 10,
+             "suppressNullValues": true
```
Comment (fhcarter) [inline objectRecognitionSource/README.md]: `suppressEmptyNeuralNetResults` or something.  "null values" is much too general.

File: objectRecognitionSource/src/test/java/io/vantiq/extsrc/objectRecognition/neuralNet/TestYoloProcessor.java [hunk 2]
```diff
@@ -80,6 +83,11 @@
     static final int KEYBOARD_CROPPED_WIDTH = 230;
     static final int KEYBOARD_CROPPED_HEIGHT = 125;
 
+    // Used to test suppressNullValues
+    static final String NULL_IP_CAMERA_ADDRESS = "http://183.77.203.213:80/-wvhttp-01-/GetOneShot?image_size=640x480&frame_count=1000000000";
```
Comment (fhcarter) [inline objectRecognitionSource/src/test/java/io/vantiq/extsrc/objectRecognition/neuralNet/TestYoloProcessor.java]: NO_RECOGNIZED_OBJECTS_CAMERA_ADDRESS? or something

**Comment by namirfa** [comment]
Just changed the code, running tests one last time and then I'll push the fix.

**Comment by namirfa** [comment]
@fhcarter 
Renamed the config option.

File: objectRecognitionSource/src/main/java/io/vantiq/extsrc/objectRecognition/ObjectRecognitionCore.java [hunk 3]
```diff
@@ -68,6 +68,7 @@
     
     public String outputDir;
     public String lastQueryFilename;
+    public Boolean suppressNullValues = false;
```
Comment (fhcarter) [inline objectRecognitionSource/src/main/java/io/vantiq/extsrc/objectRecognition/ObjectRecognitionCore.java]: name should change here...

**Pull request merged**